web_services/foxx/brightid/db.js

```diff
@@ -536,52 +617,76 @@ function getRecoveryConnections(userId) {
 function setSigningKey(signingKey, key, timestamp) {
   usersColl.update(key, {
     signingKeys: [signingKey],
-    updateTime: timestamp
+    updateTime: timestamp,
   });
 }
 
 function getSponsorship(appUserId) {
   const sponsorship = sponsorshipsColl.firstExample({ appId: appUserId });
-  if (! sponsorship) {
+  if (!sponsorship) {
     throw new errors.AppUserIdNotFoundError(appUserId);
   }
   return sponsorship;
 }
 
 function isSponsored(key) {
-  return sponsorshipsColl.firstExample({ '_from': 'users/' + key }) != null;
+  let sponsored =
+    sponsorshipsColl.firstExample({ _from: "users/" + key }) != null;
+
+  // It's a hotfix and should remove after the next client release
+  if (!sponsored) {
+    for (const context of contextsColl.all()) {
+      const coll = db._collection(context.collection);
+      if (coll) {
+        const sponsoredAppIds = query`
+          FOR c IN ${coll}
+            FILTER c.user == ${key}
+            FOR s IN ${sponsorshipsColl}
+              FILTER s.appId == c.contextId
+              RETURN s.appId
+        `.toArray();
+        if (sponsoredAppIds.length > 0) {
+          return true;
+        }
+      }
+    }
+  }
```

Comment:
This is the part I should review, right @siftal ?

Reply:
This looks good.  It could be more efficient by putting the loop on the context collections in the query, but aql has a more difficult syntax than javascript and is harder to debug, so I think this is a fine solution as a temporary fix until the client update is tested and released.